Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -761,10 +761,18 @@ def network_shutdown(self):
 
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
+
+        # if pstate can't be found
+        if not pstate.get('state', 'engineresumedata'):
```

Comment:
you mean the `pstate`? It is stored in `pstate_for_restart` in `LibtorrentDownloadImpl`. And it is updated when we get an alert.